src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -2768,6 +2760,33 @@ grpc_error *grpc_chttp2_incoming_byte_stream_finished(
   return error;
 }
 
+static void incoming_byte_stream_shutdown(grpc_exec_ctx *exec_ctx,
```

Comment:
No, I think shutdown should never be called with GRPC_ERROR_NONE.

I didn't intend the invariant here to be that *all* subsequent calls to pull() return the error, only that the *next* call to pull() returns the error.  In practice, once a caller gets an error from pull(), they're not going to call it again.  I've attempted to clarify the wording in byte_stream.h about this.